bokehjs/src/lib/models/glyphs/defs.ts

```diff
@@ -70,113 +70,145 @@ function _one_tri(ctx: Context2d, r: number): void {
   ctx.closePath()
 }
 
-function asterisk(ctx: Context2d, i: number, r: number, line: LineVector, _fill: FillVector): void {
+type VectorVisuals = {line: LineVector, fill: FillVector, hatch: HatchVector}
+
+function asterisk(ctx: Context2d, i: number, r: number, visuals: VectorVisuals): void {
   _one_cross(ctx, r)
   _one_x(ctx, r)
 
-  if (line.doit) {
-    line.set_vectorize(ctx, i)
+  if (visuals.line.doit) {
+    visuals.line.set_vectorize(ctx, i)
     ctx.stroke()
   }
 }
 
-function circle(ctx: Context2d, i: number, r: number, line: LineVector, fill: FillVector): void {
+function circle(ctx: Context2d, i: number, r: number, visuals: VectorVisuals): void {
   ctx.arc(0, 0, r, 0, 2*Math.PI, false)
 
-  if (fill.doit) {
-    fill.set_vectorize(ctx, i)
+  if (visuals.fill.doit) {
```

Comment:
This is wrong and not going to work, but there is no point in dealing with this before PR #10788 is merged (which unifies hatch pattern handling with other types visuals).